src/init.cpp

```diff
@@ -303,6 +303,7 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-bantime=<n>", strprintf(_("Number of seconds to keep misbehaving peers from reconnecting (default: %u)"), 86400));
     strUsage += HelpMessageOpt("-bind=<addr>", _("Bind to given address and always listen on it. Use [host]:port notation for IPv6"));
     strUsage += HelpMessageOpt("-connect=<ip>", _("Connect only to the specified node(s)"));
+    strUsage += HelpMessageOpt("-connect-thinblock=<ip>", _("Connect to a thinblock node(s). Blocks will only be downloaded from a thinblock peer.  If no connections are possible then regular blocks will then be downloaded form any other connected peers."));
```

Comment:
if no parameter is specified then it will download thinblocks from
thinblock supporting peers and regular blocks from others. all depending
on who has an available block to download first. 

Of course when we first roll it out there won't be many thinblock peers
out there so it's good to use this for some node operators that really
want the thinblocks with certainty or if they want to setup some kind of
thinblock relay network.

On 04/02/2016 6:05 PM, gandrewstone wrote:
>
> In src/init.cpp
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/8#discussion_r51968479>:
>
> > @@ -303,6 +303,7 @@ std::string HelpMessage(HelpMessageMode mode)
> >      strUsage += HelpMessageOpt("-bantime=<n>", strprintf(_("Number of seconds to keep misbehaving peers from reconnecting (default: %u)"), 86400));
> >      strUsage += HelpMessageOpt("-bind=<addr>", _("Bind to given address and always listen on it. Use [host]:port notation for IPv6"));
> >      strUsage += HelpMessageOpt("-connect=<ip>", _("Connect only to the specified node(s)"));
> > +    strUsage += HelpMessageOpt("-connect-thinblock=<ip>", _("Connect to a thinblock node(s). Blocks will only be downloaded from a thinblock peer.  If no connections are possible then regular blocks will then be downloaded form any other connected peers."));
>
> what happens if this parameter is not specified? I'm hoping that the
> client automatically uses thin blocks if it connects to a peer that
> supports them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/8/files#r51968479>.
>